### PR TITLE
feat: add InterRecordCumulatedDifferenceProcessor

### PR DESCRIPTION
**Pull request by olivierabz**
First of the 2 processors required by one of our client to handle variables whose value increases across records (e.g. mileage of a car)

### CODE REVIEW

File: avatars/processors/__init__.py [hunk 1]
```diff
@@ -13,5 +16,6 @@
     "PerturbationProcessor",
     "ProportionProcessor",
     "RelativeDifferenceProcessor",
-    "ToCategoricalProcessor ",
+    "ToCategoricalProcessor",
```
Comment (jpetot) [inline avatars/processors/__init__.py:19]: thanks nice fix

**Review by jpetot — APPROVED** [review]
I don't see the plus value of the processor but the implementation and the doc looks good 👍

File: avatars/processors/inter_record_cumulated_difference.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    ...    )
+    >>> processor.preprocess(df)
+       id  first_value  value_difference
+    0   1         1000              25.0
+    1   2        20000               2.0
+    2   1         1000               0.0
+    3   1         1000             105.0
+    4   2        20000               0.0
+    5   2        20000              40.0
+
```
Comment (mguillaudeux) [inline avatars/processors/inter_record_cumulated_difference.py:46]: Woaw that's a super nice trick to anonymise hierarchical data !

**Review by mguillaudeux — APPROVED** [review]
That's a very smart trick IMO ! Good job !

File: avatars/processors/__init__.py (hunk 1 above)
Comment (olivierabz) [reply avatars/processors/__init__.py:19]: > I don't see the added value of the processor

This processor is needed with hierarchical data if we have a variable like `mileage` that can only increase with time. If we leave it as is, we'll have incoherent values after avatarization. This processor expresses the value as a difference with the previous values (in other words and in the car context, the distance of a trip). The postprocess re-expresses it as car mileage.

**Pull request merged**